jooby/src/main/java/io/jooby/problem/ProblemDetailsErrorHandler.java

```diff
@@ -0,0 +1,218 @@
+/*
+ * Jooby https://jooby.io
+ * Apache License Version 2.0 https://jooby.io/LICENSE.txt
+ * Copyright 2014 Edgar Espina
+ */
+package io.jooby.problem;
+
+import edu.umd.cs.findbugs.annotations.NonNull;
+import io.jooby.*;
+import io.jooby.exception.*;
+import org.slf4j.Logger;
+
+import java.util.Arrays;
+import java.util.LinkedHashMap;
+import java.util.Map;
+
+import static io.jooby.MediaType.*;
+import static io.jooby.StatusCode.*;
+
+public class ProblemDetailsErrorHandler extends DefaultErrorHandler {
+
+  private boolean log4xxErrors;
+
+  public ProblemDetailsErrorHandler log4xxErrors() {
+    this.log4xxErrors = true;
+    return this;
+  }
+
+  @Override
+  public void apply(@NonNull Context ctx, @NonNull Throwable cause, @NonNull StatusCode code) {
+    Logger log = ctx.getRouter().getLog();
+    if (cause instanceof NotAcceptableException ex) {
+      // no matching produce type, respond in html
+      var problem = HttpProblem.valueOf(ex.getStatusCode(),
+          NOT_ACCEPTABLE.reason(),
+          "Server cannot produce a response matching the list of " +
+          "acceptable values defined in the request's 'Accept' header"
+      );
+
+      logProblem(ctx, problem, cause);
+      sendHtml(ctx, problem);
+      return;
+    }
+
+    try {
+      HttpProblem problem = evaluateTheProblem(ctx, cause, code);
+
+      logProblem(ctx, problem, cause);
+
+      MediaType type = ctx.accept(Arrays.asList(ctx.getRequestType(), html, text, json, xml));
+      ctx.setResponseCode(problem.getStatus());
+      problem.getHeaders().forEach(ctx::setResponseHeader);
+
+      if (text.equals(type)) {
+        ctx.setResponseType(text).send(problem.toString());
+      } else if (html.equals(type)) {
+        sendHtml(ctx, problem);
+      } else {
+        try {
+          setResponseType(ctx, type);
+          ctx.render(toProblemResponse(problem));
+        } catch (NotAcceptableException ex) {
+          sendHtml(ctx, problem);
+        }
+      }
+    } catch (Exception ex) {
+      log.error("Unexpected error during ProblemDetailsErrorHandler execution", ex);
+    }
+  }
+
+  private void setResponseType(Context ctx, MediaType type) {
+    if (json.equals(type)) {
+      ctx.setResponseType(PROBLEM_JSON);
+    } else if (xml.equals(type)) {
+      ctx.setResponseType(PROBLEM_XML);
+    }
+  }
+
+  private HttpProblem evaluateTheProblem(Context ctx, Throwable cause, StatusCode statusCode) {
+    HttpProblem problem;
+    if (cause instanceof HttpProblem httpProblem) {
```

Comment:
let's reuse here

Reply:
following the answer above, it will catch problems inherited from `HttpProblem` appropriately